### ISSUE TITLE
timeouts ignored creating postgresql_server 

### ISSUE THREAD

**Issue by psteinds** [terraform-providers/terraform-provider-azurerm#11568]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
As far back as:
TF: 0.14.5
Azurerm: 2.51.0
Also with the latest versions:
TF: 0.15.1
Azurerm: 2.57.0

### Affected Resource(s)

* `azurerm_postgresql_server`

### Terraform Configuration Files

```hcl
terraform {
    backend "azurerm" {
        storage_account_name = "REDACTED"
        container_name = "tfstatedefault"
        key = "dsprod_dspeu4pvault_default.terraform.tfstate"
        access_key = "REDACTED"
    }
    required_providers {
        azurerm = {
            source  = "hashicorp/azurerm"
            version = "=2.57.0"
        }
    }
}
provider "azurerm" {
    features        {}
    subscription_id = "REDACTED"
}
data "azurerm_resource_group" "dsprod_dspeu4pvault" {
  name = "dspeu4pvault"
}

data "azurerm_subnet" "eu4pkubeapps" {
  name                 = "eu4pkubeapps"
  virtual_network_name = "eu4"
  resource_group_name  = "dspeu4network"
}
data "azurerm_subnet" "eu4privatelink" {
  name                 = "eu4privatelink"
  virtual_network_name = "eu4"
  resource_group_name  = "dspeu4network"
}

resource "azurerm_key_vault" "dpeu4pvault" {
    name = "dpeu4pvault"
    resource_group_name = data.azurerm_resource_group.dsprod_dspeu4pvault.name
    location            = data.azurerm_resource_group.dsprod_dspeu4pvault.location
    enabled_for_deployment = false
    enabled_for_disk_encryption = false
    enabled_for_template_deployment = false
    tenant_id = "REDACTED"
    sku_name = "premium"
    tags = {
      createdby = "terraform"
    }
}

resource "azurerm_key_vault_access_policy" "policy_name" {
    key_vault_id = azurerm_key_vault.dpeu4pvault.id
    tenant_id = "REDACTED"
    object_id = "REDACTED"
    key_permissions = ["Get", "WrapKey", "UnwrapKey"]
    secret_permissions = ["Get"]
    storage_permissions = ["Get"]
}

resource "azurerm_postgresql_server" "dspeu4pvault" {
    timeouts {
      create = "60m"
      delete = "30m"
    }
    name = "dspeu4pvault"
    resource_group_name = data.azurerm_resource_group.dsprod_dspeu4pvault.name
    location            = data.azurerm_resource_group.dsprod_dspeu4pvault.location
    geo_redundant_backup_enabled = true
    storage_mb = "5120"
    backup_retention_days = "35"
    auto_grow_enabled = true
    sku_name                     = "GP_Gen5_2"
    administrator_login          = "psqladmin"
    administrator_login_password = "REDACTED"
    version                      = 11
    ssl_enforcement_enabled              = true
    lifecycle {
      ignore_changes = [storage_mb,geo_redundant_backup_enabled]
    }
    tags = {
      createdby = "terraform"
    }
}
```

### Expected Behaviour

resource created

### Actual Behaviour
...
azurerm_postgresql_server.dspeu4pvault: Still creating... [20m11s elapsed]
Error: waiting for creation of PostgreSQL Server "dspeu4pvault" (Resource Group "dspeu4pvault"): Code="OperationTimedOut" Message="The operation timed out and automatically rolled back. Please retry the operation."

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

The 2.0 provider supported timeouts block is used but despite being set to 60 minutes it always times out at 20 minutes.

**Comment by psteinds** [comment]
This was ran again with TF_LOG=DEBUG but perhaps I did this wrong because the only additional output I got was in this error section:
│   with azurerm_postgresql_server.dspeu4pvault,
│   on postgres.tf line 10, in resource "azurerm_postgresql_server" "dspeu4pvault":
│   10: resource "azurerm_postgresql_server" "dspeu4pvault" {

**Comment by aristosvo** [comment]
I've created a lot of `azurerm_postgresql_server` resources the last few months but never had this problem, is this showing every time you create a `azurerm_postgresql_server` resource?

This is probably Azure API behaviour showing up, based on the error output. The default timeout for `create` is already 60 minutes btw.

**Comment by psteinds** [comment]
This was only happening in the Germany west central region, we were able to create databases in other regions.  According to support this was caused by a failure to set the `geo_redundant_backup_enabled = true`.  Support has confirmed that this is a result of us having a `large storage option` turned on in that region and azure being unable to support geo redundant backup of that storage type in the Germany west central region.  Options to resolve are to not use `geo_redundant_backup` or have support disable the `large storage` flag on this region/subscription.
